pep_sphinx_extensions/pep_processor/transforms/pep_headers.py

```diff
@@ -97,20 +98,28 @@ def apply(self) -> None:
             field.parent.remove(field)
 
 
-def _list_name_from_thread(node: nodes.reference) -> nodes.raw:
+def _pretty_thread(text: nodes.Text) -> nodes.Text:
+    # we want to keep these as lowercase:
+    if "python-dev" in text:
+        return nodes.Text("python-dev")
+    if "python-committers" in text:
+        return nodes.Text("python-committers")
+
+    parts = text.split("/")
+
     # mailman structure is
     # https://mail.python.org/archives/list/<list name>/thread/<id>
+    try:
+        list_name = parts[parts.index("archives") + 2].removesuffix("@python.org")
+        return nodes.Text(list_name.title().replace("Sig", "SIG").replace("-", " "))
+    except ValueError:
+        pass
+
     # pipermail structure is
     # https://mail.python.org/pipermail/<list name>/<month-year>/<id>
-    parts = node[0].split("/")
     try:
-        list_name = parts[parts.index("archives") + 2]
-        masked_name = list_name.replace("@", "&#32;&#97;t&#32;")
+        list_name = parts[parts.index("pipermail") + 1]
+        return nodes.Text(list_name.title().replace("Sig", "SIG").replace("-", " "))
```

Comment:
```suggestion
        return nodes.Text(list_name.title().replace("Sig", "SIG"))
```

See above. Also, might be nice to wrap this repeated one-liner in a function, but your call of course.